desc/optimize/utils.py

```diff
@@ -336,3 +336,12 @@ def find_matching_inds(arr1, arr2):
         if len(j):
             inds.append(i)
     return np.asarray(inds)
+
+
+def f_where_x(x, xs, fs):
+    """Return fs where x==xs."""
```

Comment:
Could make this documentation more clear: 
`"""Returns fs[i] such that xs[i]=x."""`
or something similar. 

Also could probably remove assert statements.